tests/test_utils.py

```diff
@@ -40,12 +41,30 @@ def test_broadcast_cshapes(cshape, reference):
         assert isinstance(broadcast, type(signal))
 
 
+@pytest.mark.parametrize("shape1, shape2, caxis, reference1, reference2", [
+    ((1, 2, 3, 1), (1, 1, 2, 1), -1, (1, 2, 3), (1, 2, 2)),
```

Comment:
would be easier to understand if you passed the cshapes and added the number of samples like this
`pf.Signal(np.ones(cshape1 + (n_samples, )), 44100)`